test/browser/async_exports.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <assert.h>
+
+#include <emscripten.h>
+
+enum UnsuspendFunctionId {
+    UNSUSPEND_EXPORT = 0,
+    OTHER_UNSUSPEND_EXPORT,
+};
+
+EM_ASYNC_JS(int, suspend, (int value, int functionId), {
+  return await new Promise(resolve => {
+    window.setTimeout(() => {
+        Module['asm'][functionId === 0 ? 'unsuspendExport' : 'otherUnsuspendExport'].apply(null, [value]);
+    }, 0);
+    Module.awakeTest = resolve;
+  });
+});
+
+EM_ASYNC_JS(void, unsuspend, (int value), {
+  Module.awakeTest(value);
+});
+
+extern "C" EMSCRIPTEN_KEEPALIVE void unsuspendExport(int value) {
+    unsuspend(value);
+}
+
+extern "C" EMSCRIPTEN_KEEPALIVE void otherUnsuspendExport(int value) {
+    unsuspend(value);
```

Comment:
Maybe just make this file a `.c` file and then you don't need the `extern "C"` here.

Also the indention here looks wrong.  2 space indents everywhere is the convention I think.

Reply:
Done